### ISSUE TITLE
Import numba crashes after import s3fs

### ISSUE THREAD

**Issue by vfilimonov** [numba/numba#4751]
## Reporting a bug

- [x] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

If numba is imported after import of `s3fs`:
```python
import s3fs
import numba
```
it crashes python with an error:
```
Assertion failed: (PassInf && "Expected all immutable passes to be initialized"), function addImmutablePass, file /opt/concourse/worker/volumes/live/636f0d85-60d4-452d-50e9-ca6ab5111e2d/volume/llvmdev_1560805383854/work/lib/IR/LegacyPassManager.cpp, line 849.
Abort trap: 6
```

Import in the other way round works fine:
```python
import numba
import s3fs
```

Versions:
Python 3.7.1 (default, Oct 23 2018, 14:07:42) 
OSX: 10.14.6 (18G103)
numba: 0.46
llvmlite: 0.30.0
s3fs: 0.3.4

**Comment by vfilimonov** [comment]
It looks like `s3fs` is importing `pyarrow` under the hood:
```python
import sys
print('pyarrow' in sys.modules)
import s3fs
print('pyarrow' in sys.modules)
```
and thus this issue is related to: https://github.com/numba/numba/issues/4256

**Comment by stuartarchibald** [comment]
xref: https://github.com/numba/numba/issues/4256#issuecomment-546860012

**Comment by stuartarchibald** [comment]
Thanks for the report. Suspect that this has the same root cause as #4256

**Comment by esc** [comment]
Closing this issue as it seems to be resolved as the root cause issue at #4256 has been resolved. If this is not the case please re-open with a comment about any item that appears to be unresolved. Many thanks.
